assets/queries/k8s/containers_run_with_low_uid/query.rego

```diff
@@ -1,156 +1,37 @@
 package Cx
 
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "Pod"
-
-	spec := document.spec
-
-	isLowUID(spec)
-
-	metadata := document.metadata
-	result := checkContainersOverride(spec, "", metadata, input.document[i].id)
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "Pod"
-
-	spec := document.spec
-
-	isSetUID(spec) != true
-
-	metadata := document.metadata
-	result := checkContainersSet(spec, "", metadata, input.document[i].id)
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	spec := document.spec.jobTemplate.spec.template.spec
-
-	isLowUID(spec)
-
-	metadata := document.metadata
-	result := checkContainersOverride(spec, "spec.jobTemplate.spec.template.", metadata, input.document[i].id)
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	spec := document.spec.jobTemplate.spec.template.spec
-
-	isSetUID(spec) != true
-
-	metadata := document.metadata
-	result := checkContainersSet(spec, "spec.jobTemplate.spec.template.", metadata, input.document[i].id)
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") != "Pod"
-	object.get(document, "kind", "undefined") != "CronJob"
-
-	spec := document.spec.template.spec
-
-	isSetUID(spec) != true
-
-	metadata := document.metadata
-	result := checkContainersSet(spec, "spec.template.", metadata, input.document[i].id)
-}
+import data.generic.common as common_lib
 
 CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") != "Pod"
-	object.get(document, "kind", "undefined") != "CronJob"
-
-	spec := document.spec.template.spec
+	doc := input.document[i]
 
-	isLowUID(spec)
+	[path, value] = walk(doc)
+	securityContext := value.securityContext
 
-	metadata := document.metadata
-	result := checkContainersOverride(spec, "spec.template.", metadata, input.document[i].id)
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") != "Pod"
-	object.get(document, "kind", "undefined") != "CronJob"
-
-	spec := document.spec.template.spec
-
-	isSetUID(spec) != true
-
-	metadata := document.metadata
-	result := checkContainersSet(spec, "spec.template.", metadata, input.document[i].id)
-}
-
-#if there are no containers to override low UID setting
-checkContainersOverride(spec, path, metadata, id) = result {
-	object.get(spec, "containers", "undefined") == "undefined"
+	to_number(securityContext.runAsUser) < 10000
 
 	result := {
-		"documentId": id,
-		"searchKey": sprintf("metadata.name={{%s}}.%sspec.securityContext.runAsUser", [metadata.name, path]),
+		"documentId": doc.id,
+		"searchKey": sprintf("%s.securityContext.runAsUser=%d", [common_lib.concat_path(path), securityContext.runAsUser]),
```

Comment:
Since in line 11 we use to_number, I assume security context can be something other than an integer, maybe use to_number in the search key as well so it doesn't conflict with `%d`

Reply:
```suggestion
		"searchKey": sprintf("%s.securityContext.runAsUser=%d", [common_lib.concat_path(path), to_number(securityContext.runAsUser)]),
```

Reply:
I don't think we need to add `to_number` here, since it will only execute this line if `runAsUser` is a number, otherwise, it will fail on the first `to_number`, no?